pennylane_sf/remote.py

```diff
@@ -42,7 +44,9 @@ class StrawberryFieldsRemote(StrawberryFieldsSimulator):
     configuration file.
 
     Args:
-        wires (int): the number of modes to initialize the device in
+        wires (int, Iterable[Number, str]]): Number of subsystems accessible on the device,
+            or iterable that contains unique labels for the subsystems as numbers (i.e., ``[-1, 0, 2]``)
+            or strings (``['ancilla', 'q1', 'q2']``).
```

Comment:
```suggestion
        wires (Iterable[Number, str]]): Iterable that contains unique labels for the subsystems as numbers (i.e., ``[-1, 0, 2]``)
            or strings (``['ancilla', 'q1', 'q2']``).
```
Passing custom wires will error; `wires` is not a valid argument at the moment (nor is `kwargs` a part of the signature for `__init__`).

This could be mended by:
* Adding a keyword-only `wires` parameter to the signature
* Making `wires` **only** accept an `Iterable`: previously `wires` was not added because the number of wires is inferred from SF by specifying a specific backend
* Checking that the number of unique labels specified equals the number of modes for the backend
* Adding testing that involves mocking out `sf.RemoteEngine` and instantiates `StrawberryFieldsRemote` with unique labels

Reply:
Great, thanks for picking this up! I will try to go the `Iterable` only route. Especially for the hardware labellings that show how the first half of the wires is related to the second could be nice.

Reply:
I allowed for int wires to be consistent with other devices, but check that the integer matches the number of modes. Or would you prefer only allowing iterables? 

Even though the integer option is a dummy since only one value is allowed, it gives the user an informative message when doing it wrong...